tools/sqlite3_api_wrapper/sqlite3_udf_api/sqlite3_udf_wrapper.cpp

```diff
@@ -0,0 +1,85 @@
+#include "sqlite3_udf_wrapper.hpp"
+#include "cast_sqlite.hpp"
+#include <cassert>
+
+#include "sqlite3.h"
+
+scalar_function_t SQLiteUDFWrapper::CreateSQLiteScalarFunction(scalar_sqlite_udf_t sqlite_udf, sqlite3 *db_sqlite3,
+                                                               void *pApp) {
+	scalar_function_t udf_function = [=](DataChunk &args, ExpressionState &state, Vector &result) -> void {
+		CastSQLite::InputVectorsToVarchar(args);
+
+		// Orrify all input colunms
+		unique_ptr<VectorData[]> vec_data = args.Orrify();
+
+		// Vector of sqlite3_value for all input columns
+		vector<unique_ptr<vector<sqlite3_value>>> vec_sqlite(args.ColumnCount());
+		// Casting input data to vectors of sqlite_value
+		VectorType result_vec_type = CastSQLite::ToVectorsSQLiteValue(args, result, vec_sqlite, move(vec_data));
+		result.SetVectorType(result_vec_type);
+
+		SQLiteTypeValue res_sqlite_value_type =
+		    SQLiteTypeValue::NULL_VALUE; // to hold the last sqlite value type set by UDF
+
+		sqlite3_context context;
+		context.pFunc.pUserData = pApp; // set the function data
+
+		size_t argc = args.ColumnCount();                     // num of args for the UDF
+		vector<sqlite3_value> res_sqlite_values(args.size()); // to store the results from the UDF calls
+		vector<sqlite3_value> vec_values_to_free; // sqlite_values to free because some memory allocation has happened
+
+		// traversing the vector of sqlite values
+		for (idx_t row_idx = 0; row_idx < res_sqlite_values.size(); ++row_idx) {
+
+			sqlite3_value **argv = new sqlite3_value *[argc];
+
+			// create a tuple from sqlite_values
+			for (idx_t col_idx = 0; col_idx < argc; ++col_idx) {
+				argv[col_idx] = &(*(vec_sqlite[col_idx]))[row_idx];
```

Comment:
Perhaps you could directly construct the sqlite3_value objects in the correct row-wise format in `CastSQLite::ToVectorsSQLiteValue`? That avoids having to move them from column-wise format to row-wise format here.

Reply:
Hey Mark,
Thanks for the suggestions, I already implemented them all.

But I guess this one is a little problematic. I don't think so directly construct the sqlite3_value objects in the ToVectorsSQLiteValue is a better option because we still need to perform the tuple reconstruction.
- Constructing the tuple in the CastSQLite::ToVector may inhibit the vectorized cast to sqlite3_values.
- Another issue is the values of the argv (**argv) are pointers, so we still need to store sqlite values in some place, and then constructing the argv argument to point to the sqlite3_values.

P.S. I've already implemented such code, if you want I can share with you.

Reply:
Hey Tiago, you're correct, the fact that they are pointers complicates it somewhat and the benefits of constructing them in row-wise format in the `ToVectorsSQLiteValue` is significantly lower as we would still need to set up the pointers. I'm fine with leaving this as-is.

Could you update the PR with your new code?

Reply:
I pushed the code and the workflow was triggered on my git fork.
But it appears that the workflow on duckdb repository was not triggered automatically.